index.js

```diff
@@ -166,6 +166,56 @@ function showBadge() {
 }
 
 /**
+ * Purpose: Changes the theme based upon the value passed
+ * @param {int}     themeNum - a number passed based upon what theme button the user selected
+ */
+function changeTheme(themeNum) {
+    // set the theme slug based upon the number theme passed
+    if(themeNum == 1)
+        var personaSlug = "fox-in-snow";
+    else if(themeNum == 2)
+        var personaSlug = "cozy-fox";
+    else if(themeNum == 3)
+        var personaSlug = "shiretoko-fox";
+    // if there is no number passed, set the theme to default and return
+    else
+    {
+        LightweightThemeManager.themeChanged(null);
+        return;
+    }
+
+    // start a new XMLHTTP request with AMO to request the page where we can get the addon ID
+    var request = new XMLHttpRequest();
+    request.open("GET", "https://services.addons.mozilla.org/firefox/api/addon/" + personaSlug);
+    request.onload = function() {
+        try {
+            // get the addon ID from the page returned
+            var id = request.responseXML.documentElement.getAttribute("id");
+
+            // start a new XMLHTTP request to get the theme JSON from AMO
+            var personaRequest = new XMLHttpRequest();
+            personaRequest.open("GET", "https://versioncheck.addons.mozilla.org/en-US/themes/update-check/" + id);
+            personaRequest.onload = function() {
+                try {
+                    // get the theme JSON from the response
+                    var theme = JSON.parse(personaRequest.response);
+                    // set the theme
+                    LightweightThemeManager.themeChanged(theme);
+                }
+                catch (e) {
+                    showErrorMessage("Invalid Persona");
```

Comment:
Where is `showErrorMessage` defined? Also, you are taking in `e` but not using it.